.github/workflows/tool-tests.yml

```diff
@@ -2,6 +2,9 @@ name: Tool Tests
 
```

Comment:
Sharing some additional comments from my manual testing:

* Minor release flow seems pretty fine.

* Major release flow: 
Release branch creation works fine.
But the changes made for master branch doesn't seem to be correct. Please check with the commit that I shared before

* Getting some errors after running the script for the second time:

```
....
....
subprocess.CalledProcessError: Command 'git checkout -b release-10.2-test' returned non-zero exit status 128.
```

* For the missing/incorrect parameters, I would expect throwing errors beforehand, but for the most, we throw the error in very deeper levels (like when using github api or when splitting a string etc.).
Let's make some correctness checks for them at the very beginning of our main entrypoint.
I think we can have a function to do those checks such as `check_args` or something.

* In `initialize_env` function, we should decide which repo to clone according to `prj_name` arg. 
It can be either "citus" or "citus-enterprise".  So I think we can use `add_arg(..choices=..)` for that.

* For `--is_test` & `cherry_pick_enabled` params, I think we can use `add_arg(..action='store_true'..`, instead of requiring to pass "True" string.

* For `schema_version`, I think we should error if it's passed for major release. 
But we should be okay if it is passed or not passed for patch release.
 
* I think we should set `add_arg(..required=true`..), for `prj_ver`, `main_branch` & `gh_token`.

* We should error if either of `cherry_pick_enabled` or `earliest_pr_date` is set for major release.
But we should be okay if it is passed or not passed for patch release.

* We should error if `cherry_pick_enabled` is set to true but `earliest_pr_date` is not given.

* Can we simplify `earliest_pr_date` format, like just `'%Y.%m.%d` ? I don't think we need to know hour, minute etc. ever.

Reply:
I addressed all of them except the action="store_true" parameters because when this parameter is used, it can not be set by user. It sets a constant variable not suitable for user-defined parameters

Reply:
> it can not be set by user.

Assuming `is_test` arg, using `store_true` means:
If user doesn't add `--is_test`, then `is_test` would automatically evaluate to `false`, so we will disable test mode.
If user puts `--is_test` into the args (without any additional strings), then `argparese` should interpret `is_test` as `true`.

https://stackoverflow.com/a/8203679/13370386

Is there any reason for not using that ?

Reply:
I hadn't totally grasped the parameter. As you referenced, I understood and changed that way thanks 👍